### PR TITLE
Update references to "Enterprise" to "Enterprise Server"

### LINKED ISSUE #7728: "GitHub Enterprise" could be confusing in sign in process
**Please describe the problem you think should be solved**
When signing into GitHub using Desktop, we say, "Sign in to GitHub.com" and "Sign in to GitHub Enterprise."

However, "GitHub Enterprise" actually means "GitHub Enterprise Server," as GitHub Enterprise Cloud users should be using the "Sign in to GitHub.com" flow. See https://github.com/enterprise for more on this breakout between Cloud and Server.

**Do you have any potential solutions in mind?**

@nathos brought up the suggestion to change the enterprise sign in option to "Sign in to GitHub Enterprise Server" to make this a bit more explicit and hopefully reduce confusion.

image

### PR DESCRIPTION
**Pull request by nathos**
## Overview

**Closes #7728**

## Description

This pull request updates user-facing references to "GitHub Enterprise" to "GitHub Enterprise Server", along with select in-line comments, where appropriate. (Note that GitHub rebranded "GitHub Enterprise"—the on-premises flavor of GitHub—to ["GitHub Enterprise Server"](https://github.com/enterprise))

## Release notes

Notes: [Improved] Update UI references to `GitHub Enterprise` to `GitHub Enterprise Server`

### CODE REVIEW

File: app/src/ui/clone-repository/clone-repository.tsx [hunk 1]
```diff
@@ -218,7 +218,7 @@ export class CloneRepository extends React.Component<
           selectedIndex={this.props.selectedTab}
         >
           <span>GitHub.com</span>
-          <span>Enterprise</span>
+          <span>Enterprise Server</span>
```
Comment (nathos) [inline app/src/ui/clone-repository/clone-repository.tsx]: I'm not 100% if this is purely cosmetic 💅 or if changing this tab label has a broader impact.

**Comment by tierninho** [comment]
@nathos nice job catching all the Enterprise mentions. I think you missed only one: https://github.com/desktop/desktop/blob/f37e6bf6d9a51641f3bfdf1082a3a5b166b00e1e/app/src/ui/sign-in/sign-in.tsx#L182 anthough we could go either way here.

Otherwise LGTM. 

Note, I have had no luck signing in with Enterprise credentials in dev mode so I couldn't check everything.

File: app/src/ui/clone-repository/clone-repository.tsx (hunk 1 above)
Comment (shiftkey) [reply app/src/ui/clone-repository/clone-repository.tsx]: 👍 we use a `CloneRepositoryTab` enum to represent the selected tab, so a label change should be safe.

**Comment by shiftkey** [comment]
There's also one last `Enterprise` reference here that might need to be updated:

https://github.com/desktop/desktop/blob/46291625ec0f5e72350aa714775dca16f893ee51/app/src/ui/preferences/accounts.tsx#L35

**Comment by shiftkey** [comment]
@nathos I'd love to see this go out with our next beta - is there anything we can do to help on our end to pull this into our review queue?

**Comment by nathos** [comment]
@shiftkey I have a few more instances to update and then I can flip the "Ready for review" switch. I'll try to get those pushed up today.

File: app/src/ui/publish-repository/publish.tsx [hunk 2]
```diff
@@ -141,7 +141,7 @@ export class Publish extends React.Component<IPublishProps, IPublishState> {
           selectedIndex={this.state.currentTab}
         >
           <span>GitHub.com</span>
-          <span>Enterprise</span>
+          <span>Enterprise Server</span>
```
Comment (outofambit) [inline app/src/ui/publish-repository/publish.tsx]: should this be "GitHub Enterprise Server too"?

File: app/src/ui/terms-and-conditions/terms-and-conditions.tsx [hunk 3]
```diff
@@ -71,10 +71,11 @@ export class TermsAndConditions extends React.Component<
 
           <p>
             If you configure the Software to work with one or more accounts on
-            the GitHub.com website or with an instance of GitHub Enterprise,
-            your use of the Software will also be governed your applicable
-            GitHub.com website Terms of Service and/or the license agreement
-            applicable to your instance of GitHub Enterprise ("GitHub Terms").
+            the GitHub.com website or with an instance of GitHub Enterprise
```
Comment (outofambit) [inline app/src/ui/terms-and-conditions/terms-and-conditions.tsx]: I'm not sure we want to change the wording in the terms and conditions. @billygriffin do you know?

Comment (outofambit) [reply app/src/ui/terms-and-conditions/terms-and-conditions.tsx]: actually nvm i think this is good!

**Comment by tierninho** [comment]
LGTM 👍

File: app/src/ui/publish-repository/publish.tsx (hunk 2 above)
Comment (nathos) [reply app/src/ui/publish-repository/publish.tsx]: Agreed, this should be “GitHub Enterprise Server”

**Pull request merged**